### PR TITLE
mds: add inline flag to MDS bootstrap compat in MDSMap

### PR DESCRIPTION
**Pull request by batrick**
## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by batrick** [comment]
Example failure here: /ceph/teuthology-archive/pdonnell-2022-01-31_19:13:02-fs:upgrade-master-distro-default-smithi/6651572/teuthology.log

**Comment by batrick** [comment]
(rebase)

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2022-02-03_15:05:38-fs:upgrade-wip-pdonnell-testing-20220202.230234-distro-default-smithi/

**Comment by batrick** [comment]
@vshankar PTAL

File: qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/1-volume/4-verify.yaml [hunk 1]
```diff
@@ -3,5 +3,5 @@ tasks:
     host.a:
       - ceph fs dump
       - ceph --format=json fs dump | jq -e ".filesystems | length == 1"
-      - while ! ceph --format=json mds versions | jq -e ". | add == 4"; do sleep 1; done
+      - ceph --format=json mds versions | jq -e ". | add == 4"
```
Comment (batrick) [inline qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/1-volume/4-verify.yaml:6]: aw crap, this was unintentional... will revert.

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
New run: https://pulpito.ceph.com/pdonnell-2022-02-04_21:45:38-fs:upgrade-wip-pdonnell-testing-20220204.020350-distro-default-smithi/

failure is a known transient one.

**Review by vshankar — APPROVED** [review]

**Pull request merged**